libs/storage/src/ipfs/ipfs.service.ts

```diff
@@ -59,29 +59,27 @@ export class IpfsService {
     return Buffer.concat(chunks);
   }
 
-  public async getInfo(cid: string, checkExistence = true): Promise<BlockStatResult> {
-    if (checkExistence && !(await this.existsInLocalGateway(cid))) {
-      throw new Error('Requested resource does not exist');
-    }
-
-    this.logger.debug(`Requesting IPFS stats for ${cid}`);
-    const response = await this.ipfs.block.stat(CID.parse(cid));
-    this.logger.debug(`IPFS response: ${JSON.stringify(response)}`);
-    if (!response.cid) {
-      throw new Error('Requested resource not found');
-    }
-    return response;
-  }
-
-  public async existsInLocalGateway(cid: string): Promise<boolean> {
+  /**
+   * gets the true content-length of the CID file from the IPFS gateway
+   * if the file does not exist locally, returns -1
+   */
+  public async contentLengthInLocalGateway(cid: string): Promise<number> {
     this.logger.debug(`Requesting HEAD for ${cid}`);
     const response = await fetch(getIpfsCidPlaceholder(cid, this.gatewayUrl), {
       method: "HEAD",
       headers: {
         "Cache-Control": "only-if-cached",
       },
     });
-    return response && response.status === 200;
+    if (response && response.status === 200) {
+      return Number(response.headers.get("Content-Length"));
+    } else {
+      return -1;
```

Comment:
Hmm, since the old function was throwing an error if the resource doesn't exist, perhaps we want to preserve that behavior?

Reply:
Good call, will update.